eth/downloader/skeleton.go

```diff
@@ -648,7 +648,11 @@ func (s *skeleton) processNewHead(head *types.Header, final *types.Header, force
 	}
 	if parent := rawdb.ReadSkeletonHeader(s.db, number-1); parent.Hash() != head.ParentHash {
 		if force {
-			log.Warn("Beacon chain forked", "ancestor", parent.Number, "hash", parent.Hash(), "want", head.ParentHash)
+			ancestor := uint64(0)
+			if parent != nil {
+				ancestor = parent.Number.Uint64()
+			}
```

Comment:
This doesn't fix the issue, because is `parent` is `nil`, then it will crash on line `649` on `parent.Hash() != head.ParentHash`.

Reply:
No, that doesn't crash:

```go
var header *types.Header
fmt.Println("empty hash is", header.Hash())
```

```
empty hash is 0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347
```

Reply:
Oh wow, actually I'm wrong here
```
func TestFoo(t *testing.T) {
	var a types.Header
	fmt.Printf("Hash: %#x\n", a.Hash())
}

```
Doing a `Hash()` on `nil` is perfectly fine